ballerina/SQLClient.bal

```diff
@@ -0,0 +1,165 @@
+        sql:ParameterizedQuery params = `(`;
+        string[] keys = self.fieldMetadata.keys();
+        int columnCount = 0;
+        foreach string key in keys {
+            if self.fieldMetadata.get(key).autoGenerated {
+                continue;
+            }
+            if columnCount > 0 {
+                params = sql:queryConcat(params, `,`);
+            }
+            params = sql:queryConcat(params, `${<sql:Value>'object[key]}`);
+            columnCount = columnCount + 1;
+        }
+        params = sql:queryConcat(params, `)`);
+        return params;
+    }
+
+    private function getColumnNames() returns sql:ParameterizedQuery {
+        sql:ParameterizedQuery params = ` `;
+        string[] keys = self.fieldMetadata.keys();
+        int columnCount = 0;
+        foreach string key in keys {
+            if self.fieldMetadata.get(key).autoGenerated {
+                continue;
+            }
+            if columnCount > 0 {
+                params = sql:queryConcat(params, `, `);
+            }
+            params = sql:queryConcat(params, stringToParameterizedQuery(self.fieldMetadata.get(key).columnName));
+            columnCount = columnCount + 1;
+        }
+        return params;
+    }
+
+    // TODO: handle composite keys (record types)
+    private function getGetKeyWhereClauses(anydata key) returns sql:ParameterizedQuery|error {
+        map<anydata> filter = {};
+        filter[self.keyFields[0]] = key;
+        return check self.getWhereClauses(filter);
+    }
+
+    function getWhereClauses(map<anydata> filter) returns sql:ParameterizedQuery|error {
+        sql:ParameterizedQuery query = ` `;
+    
+        string[] keys = filter.keys();
+        foreach int i in 0 ..< keys.length(){
+            if i > 0 {
+                query = sql:queryConcat(query, ` AND `);
+            }
+            query = sql:queryConcat(query, check self.getFieldParamQuery(keys[i]), ` = ${<sql:Value>filter[keys[i]]}`);
+        }
+        return query;
+    }
+
+    function getSetClauses(record {} 'object) returns sql:ParameterizedQuery|error {
+        sql:ParameterizedQuery query = ` `;
+        string[] keys = 'object.keys();
+        foreach int i in 0 ..< keys.length(){
+            if i > 0 {
+                query = sql:queryConcat(query, `, `);
+            }
+            query = sql:queryConcat(query, check self.getFieldParamQuery(keys[i]), ` = ${<sql:Value>'object[keys[i]]}`);
+        }
+        return query;
+    }
+
+    function getFieldParamQuery(string fieldName) returns sql:ParameterizedQuery|FieldDoesNotExist {
+        FieldMetadata? fieldMetadata = self.fieldMetadata[fieldName];
+        if fieldMetadata is () {
+            return <FieldDoesNotExist>error("Field '" + fieldName + "' does not exist in entity '" + self.entityName + "'.");
+        }
+        return stringToParameterizedQuery(fieldMetadata.columnName);
+    }
+
+    function close() returns error? {
+        return self.dbClient.close();
+    }
+}
```

Comment:
```suggestion
}

```

Reply:
Fixed in [54fcb0d](https://github.com/ballerina-platform/module-ballerina-persist/pull/1/commits/54fcb0def82f78c0e312798b888dadd74e69b77d)